### ISSUE TITLE
Question - Disable field based on a value of another

### ISSUE THREAD

**Issue by dangrima90** [jarvelov/vue-form-json-schema#50]
Hi! Very good job on the library, been testing it our and it seems like an awesome tool!

I have a general question on an issue I am having. I was looking at the `displayOptions` property in the UI Schema object, and how it works is that based on the condition in this property an element is hidden or shown.

Is there a way to have a similar logic, but to set some custom property on a component? My use case is that I'd like to disable a component based on the value of another component. For example, say you have a checkbox "Yes / No" which based on that value will enable or disable another section in the form.

**Comment by jarvelov** [comment]
Hi, and thanks! Glad you like it!

No, this does not exist yet. At least not in any form I can recommend (it can be achieved today by duplicating the fields, one for each state, but this can get extremely verbose fast).

I have however wanted to get this functionality working for some time and have been experimenting with something I call `dynamicOptions`, which works just like `displayOptions`.

You can see a preview of this feature below.
![Peek 2020-04-28 01-15](https://user-images.githubusercontent.com/6322675/80430242-5f425c80-88ee-11ea-80b2-f6f66e0fd38d.gif)
 
The UI Schema would look something like this:

```js
{
  component: 'input',
  model: 'lastName',
  fieldOptions: {
    class: ['form-control'],
    on: ['input'],
    attrs: {
      placeholder: 'Last name',
    },
  },
  // Can be either an object with the following properties, or an array of objects like it
  dynamicOptions: {
    schema: {
      required: ['noLastName'],
      properties: {
        noLastName: {
          const: true,
        },
      },
    },
    // Any other field property, such as component, children etc. can be used
    fieldOptions: {
      attrs: {
        disabled: true,
      },
    },
  },
},
```

It's not ready for release yet as I have not had time to test out this feature enough yet. However, if you want I can release it in a prerelease version if you want to check it out.

**Comment by dangrima90** [comment]
Thank you for the quick response. That looks great, exactly what I had in mind!

It would be great if you could have it in a prerelease version. I could test it out a bit and let you know if I find any issues.

**Comment by jarvelov** [comment]
You can now install the `2.5.4-beta.0` using `npm install vue-form-json-schema@2.5.4-beta.0`

Here's a working demo of the GIF above: https://codesandbox.io/s/issue-50-1txyz

Test it out and let me know if you run into any problems.

**Comment by dangrima90** [comment]
Great, will let you know of any issues I find.

**Comment by jarvelov** [comment]
@dangrima90 The dynamic properties has just been released in `v2.6.0`, however the syntax has changed a bit.

In the pre-release you would do:

```js
{
  component: 'input',
  dynamicOptions: {
    schema: {
       ...
    },
    // Any other field property, such as component, children etc. can be used
    fieldOptions: {
      attrs: {
        disabled: true,
      },
    },
  },
},
```

However now all properties which should be dynamically updated have to be nested under an `options` property.

```js
{
  component: 'input',
  dynamicOptions: {
    schema: {
       ...
    },
    options: {
      // Any other field property, such as component, children etc. can be used
      fieldOptions: {
        attrs: {
          disabled: true,
        },
      },
    },
  },
},
```

**Comment by dangrima90** [comment]
@jarvelov thanks for the update, very much appreciated.
